lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -579,15 +580,19 @@ def _retry(self, job_id: str, job: Job, parent_job: Job, as_admin: bool = False)
                     exception=e,
                 )
 
-        # 2) Notify the retry_parent that it has been retried
+        # 2) Notify the retry_parent that it has been retried by adding a retry id
         try:
-            job.modify(inc__retry_count=1)
+            job.modify(push__retry_ids=retry_job_id)
```

Comment:
How difficult would it be to add a test where this fails, showing that the retry toggle is still false? That's a key piece of behavior. You might need to add a SDKMR.modify_job() method so the modify can be mocked.

Maybe also failing on the parent job update as well.

Reply:
On second thought a mock test wouldn't really be useful here. It'd need to be an integration test with mongo and I don't see a way to do that test cleanly.